### PR TITLE
Add non-developer provisioning

### PR DESCRIPTION
**Pull request by BaselAl**
Add separate [brew bundles](https://github.com/Homebrew/homebrew-bundle) for software for developers and non-developers.

This would allow easier provisioning for employees' computers depending on their needs.

Developers would typically install programming languages, command line tools, code editors, debuggers. While non-developers would only need a consistent set of tools and services the company uses.

### CODE REVIEW

File: brew.sh [hunk 1]
```diff
@@ -3,15 +3,30 @@
 # Install Brew if not installed
 command -v brew > /dev/null || /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
 
-# Update Brew and Formulae
+# Update Brew and Formula
```
Comment (hkdobrev) [inline brew.sh:6]:
```suggestion
# Update Brew and Formulae
```

Formulae is the correct spelling of the plural form of formula. See https://formulae.brew.sh/

File: Developer [hunk 2]
```diff
@@ -0,0 +1,90 @@
+#######################################
+# Developer file for host development machine
+#######################################
```
Comment (hkdobrev) [inline Developer]:
```suggestion
###############################################
# Developer file for host development machine #
###############################################
```

😄

File: README.md [hunk 3]
```diff
@@ -1,12 +1,12 @@
-# Provision OSX
+# Provision OSX - Developer/Non-Developer
 
 Provision a new OSX machine with dev tools.
 
 ## Intro
 
-This is not a dotfiles configuration repository, nor it is a recipe repository. It is just a simple set of commands and tools to set up a developer's machine. Its aim is to reduce the initial time of installing tools and apps on a new machine.
+This is not a dotfile configuration repository, nor it is a recipe repository. It is just a simple set of commands and tools to set up a developer's/non-developer machine. Its aim is to reduce the initial time of installing tools and apps on a new machine.
```
Comment (hkdobrev) [inline README.md]:
```suggestion
This is not a dotfiles configuration repository, nor it is a recipe repository. It is just a simple set of commands and tools to set up a developer's/non-developer machine. Its aim is to reduce the initial time of installing tools and apps on a new machine.
```

dotfiles is an adjective in this sentence. See https://dotfiles.github.io/

File: brew.sh [hunk 4]
```diff
@@ -3,15 +3,30 @@
...
 
 # Install brew bundle
 brew tap Homebrew/bundle
 
-# Install tools and apps from the Brewfile wih Brew and Cask
-brew bundle --file=Brewfile
-pecl install xdebug
+# Install tools and apps for Devolepers or Non-Developers
+echo "Type the job role ( Developer or Non-Developer ), followed by [ENTER]:"
+read role
```
Comment (hkdobrev) [inline brew.sh]: How would you run this in non-interactive mode? Shouldn't we read an argument or option from the CLI instead? This way we can just say "run this command" or write a script wrapped in an app than runs it without a prompt.

File: brew.sh [hunk 5]
```diff
@@ -3,15 +3,30 @@
...
+  echo "Please type the correct format"
+  read role
+done
+
+if [ "$role" != "Non-Developer" ]
+then
+  brew bundle --file=Developer
+  pecl install xdebug
+else
+  brew bundle --file=Non-Developer
```
Comment (hkdobrev) [inline brew.sh]: We could probably add an extension to these files as otherwise they would usually be considered readmes or makefiles. E.g. `Developer.bundle` and `Non-Developer.bundle`.

File: brew.sh [hunk 6]
```diff
@@ -3,15 +3,30 @@
...
+  pecl install xdebug
```
Comment (hkdobrev) [inline brew.sh]:
```suggestion
```

We don't need xdebug for non-developers surely.

**Review by hkdobrev — CHANGES_REQUESTED** [review]
Initial review

**Comment by hkdobrev** [comment]
@BaselAl Please consider whether we need to make this repo private if we are to customise with more internal tools or if there would be anything sensitive. Originally, it was meant to be public so we can send it to people before they start at Clippings and they would always provision their software themselves. Now, this is no longer the case as often.

**Comment by hkdobrev** [comment]
@BaselAl Take note on how I've modified the title to be sentence-case, assigned yourself, added a label and *most importantly a description*.

File: Developer [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+cask 'google-chrome'
+
+# Google Drive
+cask 'google-drive-file-stream'
+
+# Communication
+cask 'slack'
+
+# UI & vectors
+cask 'sketch'
```
Comment (hkdobrev) [inline Developer]: Let's remove Sketch. We use Figma now. We can add the desktop tool of Figma if it's available as a cask. But if not, it's no big deal, most of the time the web version is good for browsing.

File: Developer [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+
+# Virtualisation
+cask 'vagrant'
+cask 'virtualbox'
+
+# Shared team passwords
+cask '1password'
+
+# Wiki
+cask 'notion'
```
Comment (hkdobrev) [inline Developer]:
```suggestion
```

We need to remove Notion. We use Confluence now, it doesn't have a desktop tool.

File: Developer [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+# Communication
+cask 'slack'
+
+# UI & vectors
+cask 'sketch'
+
+# Text-editors
+brew 'vim'
+tap 'neovim/neovim'
+brew 'neovim'
```
Comment (hkdobrev) [inline Developer]:
```suggestion
```
Let's remove neovim. If someone really needs it, they can install it, but it creates more problems than it solves.

File: Non-Developer [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+# Browsers
+cask 'google-chrome'
+cask 'firefox'
+
+# Communication
+cask 'slack'
+cask 'zoomus'
+
+#Storage & Secure
```
Comment (hkdobrev) [inline Non-Developer]:
```suggestion
# Storage & Secure
```
💅

**Review by hkdobrev — CHANGES_REQUESTED** [review]

File: brew.sh [hunk 11]
```diff
@@ -3,15 +3,30 @@
...
+while [[ "$role" != "Developer" && "$role" != "Non-Developer" ]]
+do
+  echo "Please type the correct format"
+  read role
+done
+
+if [ "$role" != "Non-Developer" ]
+then
+  brew bundle --file=Developer
+  pecl install xdebug
```
Comment (hkdobrev) [inline brew.sh]: We can probably move the Pecl installs to a separate script. We'd also need to install Composer (explicitly not via Brew). See https://getcomposer.org/doc/faqs/how-to-install-composer-programmatically.md This could be in another PR, of course.

File: brew.sh (hunk 1 above)
Comment (BaselAl) [reply brew.sh:6]: Thank you Harry. I didn't know that "Formulae" is an alternative plural of formula. Anyway It means the same thing as "Formulas" but how on earth I missed "s".

Comment (hkdobrev) [reply brew.sh:6]: Brew uses the more scientific and British-English formulae.

Comment (BaselAl) [reply brew.sh:6]: By the way this comment "#Update Brew and Formulae" and this action "brew update" means to me something totally different. I wasn't aware what is the purpose of brew formulae:

So now I know the meaning:
A formula provides instructions and metadata for Homebrew to install a piece of software. Every Homebrew formula is a Formula.

File: Developer (hunk 2 above)
Comment (BaselAl) [reply Developer]: Honestly it is much better 😊

File: brew.sh (hunk 1 above)
Comment (hkdobrev) [reply brew.sh:6]: Yup, Brew has 2 components - package manager and a registry of instructions.

File: README.md (hunk 3 above)
Comment (BaselAl) [reply README.md]: I'm so embarrassed. I will never work late at night again. I am sorry for the awful change.

Comment (hkdobrev) [reply README.md]: @BaselAl Hey, I agree with not working late at night, but this change is an innocuous one.

In code reviews, my intention here is not to make others feel bad at all. I'm trying to be thorough in my code reviews in order to help others improve and teach them catch things themselves. Everything should be taken as a suggestion and best argument wins. Also, everyone makes mistakes and even a lot of my PR comments here and elsewhere may not be entirely correct, but if they trigger the right provocation or discussion so we improve our software or learn something ourselves, then it was worth it. Check out these articles mentioned here: https://github.com/clippings/clippings/blob/master/.github/CONTRIBUTING.md#code-review

Comment (BaselAl) [reply README.md]: @hkdobrev I am very gratefull for your suggestions. I really appreciate the provided guide for reviewing code.

File: brew.sh (hunk 4 above)
Comment (BaselAl) [reply brew.sh]: @hkdobrev I really like your comment. Good call! 
Actually the idea was to use interactive mode. That is why the script asks "Type the job role ( Developer or Non-Developer ). Can I leave the implementation аs is or to creat two ./brew.sh like developer.sh and non-developer.sh ?

File: brew.sh (hunk 5 above)
Comment (BaselAl) [reply brew.sh]: @hkdobrev Good call! I will make that change.

File: brew.sh (hunk 6 above)
Comment (BaselAl) [reply brew.sh]: @hkdobrev You are right. I missed to remove it. Thank you.

File: Developer (hunk 7 above)
Comment (BaselAl) [reply Developer]: @hkdobrev Sounds good. There is a Figma formula and I will make a change.

File: Developer (hunk 8 above)
Comment (BaselAl) [reply Developer]: @hkdobrev I will make a change.

File: Developer (hunk 9 above)
Comment (BaselAl) [reply Developer]: @hkdobrev I didn't know that. Can you please clarify what problems ?

File: Non-Developer (hunk 10 above)
Comment (BaselAl) [reply Non-Developer]: 😄 @hkdobrev Thanks

File: brew.sh (hunk 11 above)
Comment (BaselAl) [reply brew.sh]: @hkdobrev Can you please clarify. Do you want me to create for example composer.sh and use the provided code.

File: brew.sh (hunk 4 above)
Comment (hkdobrev) [reply brew.sh]: @BaselAl My question is why the idea was to use interactive mode? If people run it themselves, they don't need that prompt. They either need one or other and for non-technical people is better to not have many prompts on the  command line.

File: Developer (hunk 9 above)
Comment (hkdobrev) [reply Developer]: @BaselAl Mostly that neovim is not used and is not as easy to configure. 

Even if vim is not widely used, this formula replaces the normal vi in MacOS by default with a newer version of vim. However, since we don't have active neovim users, there's no need to overcomplicate things. Power vim configuration users, can install neovim additionally.

File: brew.sh (hunk 4 above)
Comment (BaselAl) [reply brew.sh]: @hkdobrev Because I remember that we discussed something similar.

Comment (hkdobrev) [reply brew.sh]: @BaselAl I think when we've discussed it, I at least meant options (as in  command-line options) and you might have thought of options to manually choose. I even think, it could be 2 separate scripts that run a common one with a different argument/option. This way we could point a specific person to a specific file to run.

File: brew.sh (hunk 1 above)
Comment (hkdobrev) [reply brew.sh:6]: This suggestion still stands.

File: Developer (hunk 2 above)
Comment (hkdobrev) [reply Developer]: We can either unify it like that in the new files or just simplify it all and leave a single `######` line under the text like Markdown.

File: brew.sh (hunk 1 above)
Comment (hkdobrev) [reply brew.sh:6]: This suggestion still stands.

File: Developer (hunk 2 above)
Comment (hkdobrev) [reply Developer]: This suggestion still stands, can we just simplify the sections so we don't bother in the future?

File: non-developer.sh [hunk 12]
```diff
@@ -3,15 +3,14 @@
 # Install Brew if not installed
 command -v brew > /dev/null || /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
 
-# Update Brew and Formulae
+# Update Brew and Formula
```
Comment (hkdobrev) [inline non-developer.sh:6]:
```suggestion
# Update Brew and Formulae
```

**Review by hkdobrev — CHANGES_REQUESTED** [review]

File: Developer (hunk 9 above)
Comment (hkdobrev) [reply Developer]: This suggestion still stands.

Comment (BaselAl) [reply Developer]: Hi @hkdobrev   
I've done it.

Comment (hkdobrev) [reply Developer]: @BaselAl You can then resolve the conversation :)

File: non-developer.sh (hunk 12 above)
Comment (hkdobrev) [reply non-developer.sh:6]: This suggestion still stands.

**Review by hkdobrev — APPROVED** [review]

**Review by pmilev — APPROVED** [review]
